### ISSUE TITLE
Torch.sort slower than nn.Conv2d.

### ISSUE THREAD

**Issue by timmywanttolearn** [pytorch/pytorch#76654]
### 🐛 Describe the bug

At cpu, `torch.sort` is so slow that it is even slower than convulutional layers. I remembered that `torch.sort` is O(Nlog(N)) complexity. Convolutional layer should be heavier than that.
Here is the code
```
 
import torch
import torch.nn as nn
import torch.multiprocessing as mp
import torch.distributed as dist
import torch.nn.functional as F
from torchvision.models import mobilenet_v2
import time
input = torch.rand([64,3,224,224])
conv2d = torch.nn.Conv2d(3,32,(3,3),stride= (2,2), padding = (1,1),bias= False)
time_sort_avg = 0.0
time_conv_avg = 0.0
for i in range(2):
    print(i)
    start =time.time()
    output = conv2d(input)
    time_conv = time.time() -start
    output = output.view(-1)
    start =time.time()
    sorted,index = torch.sort(output)
    time_sort = time.time() -start
    time_sort_avg += time_sort
    time_conv_avg += time_conv
print("sort time avg:",time_sort_avg/2)
print("conv time avg:",time_conv_avg/2)
```
The reslut on my cpu is 
```
sort time avg: 3.3030941486358643
conv time avg: 0.06574046611785889
```
Why is this happening?

### Versions

pytorch 1.12.0

**Comment by nikitaved** [comment]
This happening because your sort the WHOLE tensor and it does that in optimal time using `std::sort`. The caveat is that this sort is serial.

**Comment by timmywanttolearn** [comment]
> This happens because the WHOLE tensor is sorted with `std::sort` optimally. The caveat is that this sort is serial. Could you please open a feature request for the no-param `sort` to exploit parallelization? In general, your best bet is to sort along a specific dimension. Then if that dimension is not huge and the remaining dimensions will have more than `>33k` elements, you will see a nice speed-up on a multi-core machine.

Sure! I opened here.
https://github.com/pytorch/pytorch/issues/76655

**Comment by timmywanttolearn** [comment]
> This happens because the WHOLE tensor is sorted with `std::sort` optimally. The caveat is that this sort is serial. Could you please open a feature request for the no-param `sort` to exploit parallelization? In general, your best bet is to sort along a specific dimension. Then if that dimension is not huge and the remaining dimensions will have more than `>33k` elements, you will see a nice speed-up on a multi-core machine.

Sorry to trouble you again. Is there any way to execute `torch.kthvalue` parallel in one dimention ? Just like the code below
```
src,index = torch.kthvalue(input_tensor,(2,4,8))
```
to get the 2nd, 4th, and 8th smallest element in the tensor.

**Comment by nikitaved** [comment]
No, unfortunately, for now :(

**Comment by timmywanttolearn** [comment]
> No, unfortunately, for now :( You can do the trick, that is, if it is huge, you reshape it into a multi-dim tensor and apply this operation over these dimensions. Then rinse and repeat until you get the desired result.

Thank you!
